### ISSUE TITLE
Intrinsics/Builtins: Support for custom argument names/order

### ISSUE THREAD

**Issue by PhilippvK** [tum-ei-eda/seal5#151]
As far as I can tell the order and/or names of arguments used when defining and intrinsics (in the YAML settings) should match the assembly operands of the instruction definition.

The spec of the Core-V Extension builtins uses custom argument names (i.e. `i,j` instead of `rs1,rs2`) and places the "destination & accumulator" registers not at the front. Here is a concrete Examples: 

* `int32_t __builtin_riscv_cv_mac_mac (int32_t x, int32_t y, int32_t z)` ([Source](https://github.com/openhwgroup/core-v-sw/blob/master/specifications/corev-builtin-spec.md))

  _Argument/result mapping:_
  - result, z: `rD`
  - x: `rs1`
  - y: `rs2`

  _Generated assembler:_
  ```gas
  cv.mac  rD,rs1,rs2
  ```

We should...
- [ ] Provide (optional) YAML syntax to define how the builtins' argument names map to the instructions' operands
- [ ] Make sure that the order of arguments is generated as expected
- [ ] Clarify (docstring/readme) how argument names of intrinsics are specified implicitly/explicitly

**Comment by thomasgoodfellow** [comment]
Agreed - as it happens the previous (phase 1) implementation for the assembler faced a similar problem for instructions with unconventional argument ordering. That solution was a hack; something neater needed here.
